schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/SchemaVersionLifecycleManager.java

```diff
@@ -616,6 +617,25 @@ public SchemaVersionInfo getSchemaVersionInfo(SchemaVersionKey schemaVersionKey)
         return schemaVersionInfoCache.getSchema(SchemaVersionInfoCache.Key.of(schemaVersionKey));
     }
 
+    public SchemaVersionInfo findSchemaVersionInfoByFingerprint(final String fingerprint) throws SchemaNotFoundException {
+        final List<QueryParam> queryParams = Collections.singletonList(new QueryParam(SchemaVersionStorable.FINGERPRINT, fingerprint));
+
+        final Collection<SchemaVersionStorable> schemas = storageManager.find(SchemaVersionStorable.NAME_SPACE, queryParams);
+
+        if (schemas.isEmpty()) {
+            throw new SchemaNotFoundException(String.format("No schema found for fingerprint: %s", fingerprint));
+        } else {
+            if (schemas.size() > 1) {
+                LOG.warn(String.format("Multiple schemas found for the same fingerprint: %s", fingerprint));
+            }
+
+            return schemas.stream()
```

Comment:
Instead of sorting the schemas in-memory, can we get sorted schemas from the database and pick the first one? You can have a look at [this](https://github.com/hortonworks/registry/blob/77932b984c32f05cf17957ee0c54b185b302659d/schema-registry/core/src/main/java/com/hortonworks/registries/schemaregistry/DefaultSchemaRegistry.java#L744) code which does this.